go/libraries/doltcore/env/actions/commit.go

```diff
@@ -130,13 +130,13 @@ func CommitStaged(ctx context.Context, dEnv *env.DoltEnv, props CommitStagedProp
 		}
 	}
 
-	h, err := dEnv.UpdateStagedRoot(ctx, srt)
+	h, err := writer.UpdateStagedRoot(ctx, srt)
```

Comment:
I'm actually a little confused about how this would work. Saving the hash in the UpdateStagedHoot requires a fs passed in through the dEnv that is not part of the RepoState.

Reply:
Is this is a blocker?